### ISSUE TITLE
In Child Gallery, standardize size of child profile design so they always appear in straight line across

### ISSUE THREAD

**Issue by bchiaruttini** [mareinc/mare#138]

**Comment by joesmare** [comment]
This is no longer an issue

**Comment by joesmare** [comment]
Reopening: This is not an image issue; rather this is about siblings with names that have an extra line of text that throw off the linear presentation of the images.

**Comment by joesmare** [comment]
<img width="232" alt="2-Line Name Example" src="https://user-images.githubusercontent.com/41170314/54706593-a8ce4e80-4b15-11e9-854f-25a04f2a2d75.png">

**Comment by joesmare** [comment]
This issue can be resolved by staging the name(s) line to default to two lines of text regardless of the character needs.

**Comment by n-devr** [comment]
OK @bchiaruttini I've implemented a fix to ensure there's enough space to display the child name(s) on two lines for all cards in the child gallery, which resolves many of the alignment issues we were seeing.

Unfortunately while I was testing and debugging the issue, I noticed a couple more cases that can cause the cards to become misaligned.

1. The Ages & Legal Status text can also wrap to two lines at certain screen sizes:
 ![image](https://user-images.githubusercontent.com/5726612/81204473-3d547400-8f97-11ea-89b0-9da61c4ce7af.png)

2. Some child images (seems to only be single-child not sib groups, but I haven't confirmed that) do not get the correct height:
  
![image](https://user-images.githubusercontent.com/5726612/81205424-b0121f00-8f98-11ea-8de4-3830f8eaf5cf.png)
  Note: This doesn't happen with placeholder images, I just swapped out the real images to protect the child's identity.  I suspect this has something to do with the aspect ratio of the child image that's uploaded via the admin UI, but I haven't been able to determine a root cause yet.

There may be additional edge cases where the grid alignment gets thrown off as well, but those were the only ones I saw during my testing.  Do you want me to take care of those as part of this story as well?  While they're not directly related code-wise, I feel like they're all part of achieving the same goal (consistent alignment across all profiles in the gallery).

**Comment by bchiaruttini** [comment]
@n-devr thanks for update. Yes let's take care of all issues as part of story. @jeremysMARE might have more insight on the aspect ratio as I recall this being an ongoing coversation between he and Jared.

**Comment by jeremysMARE** [comment]
@n-devr @bchiaruttini If images are uploaded correctly, then it doesn't cause an issue. The root of this issue currently is when there are so many sibling names that the card description goes to more lines than normal. That's what causes the spacing issue.

**Comment by n-devr** [comment]
@jeremysMARE you're correct that the impetus for this story was the child names wrapping to multiple lines, but I've addressed that issue and have pushed a fix to develop.  Here's an example of what the cards look with names that wrap now:
![image](https://user-images.githubusercontent.com/5726612/81215607-97f5cc00-8fa7-11ea-9df9-87bfc4981485.png)

While addressing the initial issue I uncovered two separate issues that I outlined in my comment above.  I'm not certain why particular images are not getting the correct height, my guess is the aspect ratio.  What are the criteria for an image being uploaded correctly?  Knowing the constraints you're using will help me diagnose the root cause for the image height issue.

Here are the steps to reproduce the image height issue in dev:

1. Navigate to the [child gallery](http://development.adoptions.io/waiting-child-profiles#gallery)
2. Click modify search
3. Select only 'female' in the gender section
4. Set the maximum number of children to 1
5. You should see the issue in the first row or close to it (Isabella's profile seems to be causing the issue)
6. I was also able to reproduce this behavior with male child profiles (Jasper's profile, in this case).

Please let me know if you suspect that the images themselves are the issue in these cases.  There are safeguards we can implement in CSS to help prevent issues like this, but I'm not convinced it's the images themselves - interested to hear your feedback!

**Comment by jeremysMARE** [comment]
@n-devr Sorry about that, I was following through email and didn't see your original explanation. 

The issue with Isabella is in fact an incorrectly cropped image that was uploaded. All images have to be uploaded at a 400 x 430 aspect ratio. Her photo is incorrectly cropped to 400 x 400. It is the same case with Jasper.

If you look now, you will see both are fixed. I simply cropped the photo correctly and re-uploaded.

**Comment by n-devr** [comment]
No worries @jeremysMARE - glad to hear the images are still behaving as expected, I didn't have a reasonable backup theory if it wasn't an aspect ratio issue.  

As I mentioned I could probably write some CSS that would enforce the desired aspect ratio even if the images themselves are not quite the right size.  We'd have to decide if we want to crop or resize images that don't fit the aspect ratio, so it could lead to some cases where incorrectly sized images get cut off or distorted, but it would fix the alignment issue.  Is that something you think would be useful?

**Comment by jeremysMARE** [comment]
Yes @n-devr , I think it would be useful to have a safeguard. It is better to have systematic uniformity for the page, with the possibility that one individual photo is presented incorrectly. It is more likely that someone will notice and raise a flag for it be fixed in that case as well. 

Off the top of my head, I would guess that resizing is probably the better solution.

**Comment by n-devr** [comment]
Understood, I think I can make that work!

My tasks as I see them now are:
 - [ ] Fix the spacing issues when the ages and legal status text wraps to multiple lines
 - [ ] Ensure that images are scaled to the desired aspect ratio even if they are not natively the correct size.  Do not crop the images, rather scale them to ensure they can be fully displayed in the space available.

@jeremysMARE and @bchiaruttini please let me know if I missed anything.

**Comment by n-devr** [comment]
While I was able to fix the spacing issues with the legal status and age text, the image aspect ratio fix presented a bit of an issue.  There are scripts that ensure privacy and update the styles of the cards at runtime, and the aspect ratio solution is causing issues with that behavior.  Given that the image sizing has not presented a noticeable issue to date, I think it's better to leave them as is for the time being.

If this is something you feel strongly about I can spend the time to dig in and make an update, but otherwise I'd recommend not messing with it.
